src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private int indexOfLastAddedElement = 0;
+    private Object[] array = new Object[10];
```

Comment:
```suggestion
    private T[] array;
```

Reply:
It's impossible to initialize an array with a given type T

Reply:
You initialize array of objects, then cast it to T